### ISSUE TITLE
The `punycode` module is deprecated

### ISSUE THREAD

**Issue by Fil** [observablehq/framework#626]
```
DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
```

**Comment by Fil** [comment]
I investigated this a bit more today, since it seems that it is fixed in yarn:
https://github.com/yarnpkg/yarn/issues/9013

I upgraded yarn, but still saw the warning; however running with 
`❯ NODE_OPTIONS='--trace-deprecation' yarn dev --no-open`
showed that the error was now coming from 
`node_modules/markdown-it/lib/index.js:14:20`

fixing it then showed the same issue coming from a transitive dependency psl (jsdom#tough-cookie#psl):
`node_modules/psl/index.js:5:16`

fixing that, and the warning was gone.

To sum up, three of our dependencies will need an upgrade: yarn, markdown-it, and psl.

The fix in both cases was to transform `require("punycode")` into `require("punycode/")`; I haven't tried to understand what it did or why.

**Comment by wetlandscapes** [comment]
Just an FYI, I get the "punycode" warning when pulling the official node.js docker image (v21, Debian "bookworm"). I tried updating yarn before creating the observable project, but that didn't seem to help.
